app/models/message.rb

```diff
@@ -83,4 +86,43 @@ def update_assistant_on_conversation
     return if conversation.assistant == assistant
     conversation.update!(assistant: assistant)
   end
+
+  def post_process_markdown
+    # Your logic to process the Markdown
+    cleaned_markdown, image_urls = extract_images_and_clean_markdown(self.content_text)
+
+    # Use update_columns to avoid triggering callbacks
+    self.update_columns(content_text: cleaned_markdown)
+
+    # Attach images
+    attach_images(image_urls) if image_urls.any?
+  end
+
+  def extract_images_and_clean_markdown(markdown)
+    # Regular expression for matching Markdown image syntax ![alt](url)
+    image_regex = /!\[.*?\]\((.*?)\)/
+
+    # Extract image URLs
+    image_urls = markdown.scan(image_regex).flatten
+
+    # Remove image syntax from markdown
+    cleaned_markdown = markdown.gsub(image_regex, '')
+
+    # Return the cleaned markdown and the image URLs
+    return cleaned_markdown.strip, image_urls
+  end
+
+  def attach_images(image_urls)
+    p "attaching images: #{image_urls}"
+    document = Document.build(message: self)
+    p document
+
+    image_urls.each do |image_url|
```

Comment:
This doesn't make sense yet to iterate over multiple image_urls and then only attaching one... WIP WIP WIP :-D